core/src/contracts/update.rs

```diff
@@ -0,0 +1,157 @@
bmV3IHBvb2wgYm94IHNjcmlwdCkKCiAgdmFsIHBvb2xORlQgPSBmcm9tQmFzZTY0KCJSeXRMWWxCbFUyaFdiVmx4TTNRMmR6bDZKRU1tUmlsS1FFMWpVV1pVYWxjPSIpIC8vIFRPRE8gcmVwbGFjZSB3aXRoIGFjdHVhbCAKCiAgdmFsIGJhbGxvdFRva2VuSWQgPSBmcm9tQmFzZTY0KCJQMFFvUnkxTFlWQmtVMmRXYTFsd00zTTJkamw1SkVJbVJTbElRRTFpVVdVPSIpIC8vIFRPRE8gcmVwbGFjZSB3aXRoIGFjdHVhbCAKCiAgdmFsIG1pblZvdGVzID0gNiAvLyBUT0RPIHJlcGxhY2Ugd2l0aCBhY3R1YWwKICAKICB2YWwgcG9vbEluID0gSU5QVVRTKDApIC8vIHBvb2wgYm94IGlzIDFzdCBpbnB1dAogIHZhbCBwb29sT3V0ID0gT1VUUFVUUygwKSAvLyBjb3B5IG9mIHBvb2wgYm94IGlzIHRoZSAxc3Qgb3V0cHV0CgogIHZhbCB1cGRhdGVCb3hPdXQgPSBPVVRQVVRTKDEpIC8vIGNvcHkgb2YgdGhpcyBib3ggaXMgdGhlIDJuZCBvdXRwdXQKCiAgLy8gY29tcHV0ZSB0aGUgaGFzaCBvZiB0aGUgcG9vbCBvdXRwdXQgYm94LiBUaGlzIHNob3VsZCBiZSB0aGUgdmFsdWUgdm90ZWQgZm9yCiAgdmFsIHBvb2xPdXRIYXNoID0gYmxha2UyYjI1Nihwb29sT3V0LnByb3Bvc2l0aW9uQnl0ZXMpCiAgCiAgdmFsIHZhbGlkUG9vbEluID0gcG9vbEluLnRva2VucygwKS5fMSA9PSBwb29sTkZUCiAgCiAgdmFsIHZhbGlkUG9vbE91dCA9IHBvb2xJbi5wcm9wb3NpdGlvbkJ5dGVzICE9IHBvb2xPdXQucHJvcG9zaXRpb25CeXRlcyAgJiYgLy8gc2NyaXB0IHNob3VsZCBub3QgYmUgcHJlc2VydmVkCiAgICAgICAgICAgICAgICAgICAgIHBvb2xJbi50b2tlbnMgPT0gcG9vbE91dC50b2tlbnMgICAgICAgICAgICAgICAgICAgICAgJiYgLy8gdG9rZW5zIHByZXNlcnZlZAogICAgICAgICAgICAgICAgICAgICBwb29sSW4uY3JlYXRpb25JbmZvLl8xID09IHBvb2xPdXQuY3JlYXRpb25JbmZvLl8xICAgICYmIC8vIGNyZWF0aW9uIGhlaWdodCBwcmVzZXJ2ZWQKICAgICAgICAgICAgICAgICAgICAgcG9vbEluLnZhbHVlID09IHBvb2xPdXQudmFsdWUgICAgICAgICAgICAgICAgICAgICAgICAmJiAvLyB2YWx1ZSBwcmVzZXJ2ZWQgCiAgICAgICAgICAgICAgICAgICAgIHBvb2xJbi5SNFtMb25nXSA9PSBwb29sT3V0LlI0W0xvbmddICAgICAgICAgICAgICAgICAgJiYgLy8gcmF0ZSBwcmVzZXJ2ZWQgIAogICAgICAgICAgICAgICAgICAgICBwb29sSW4uUjVbSW50XSA9PSBwb29sT3V0LlI1W0ludF0gICAgICAgICAgICAgICAgICAgICYmIC8vIGNvdW50ZXIgcHJlc2VydmVkCiAgICAgICAgICAgICAgICAgICAgICEgKHBvb2xPdXQuUjZbQW55XS5pc0RlZmluZWQpCgogIAogIHZhbCB2YWxpZFVwZGF0ZU91dCA9IHVwZGF0ZUJveE91dC50b2tlbnMgPT0gU0VMRi50b2tlbnMgICAgICAgICAgICAgICAgICAgICAmJgogICAgICAgICAgICAgICAgICAgICAgIHVwZGF0ZUJveE91dC5wcm9wb3NpdGlvbkJ5dGVzID09IFNFTEYucHJvcG9zaXRpb25CeXRlcyAmJgogICAgICAgICAgICAgICAgICAgICAgIHVwZGF0ZUJveE91dC52YWx1ZSA+PSBTRUxGLnZhbHVlICAgICAgICAgICAgICAgICAgICAgICAmJgogICAgICAgICAgICAgICAgICAgICAgIHVwZGF0ZUJveE91dC5jcmVhdGlvbkluZm8uXzEgPiBTRUxGLmNyZWF0aW9uSW5mby5fMSAgICAmJgogICAgICAgICAgICAgICAgICAgICAgICEgKHVwZGF0ZUJveE91dC5SNFtBbnldLmlzRGVmaW5lZCkgCgogIGRlZiBpc1ZhbGlkQmFsbG90KGI6Qm94KSA9IGlmIChiLnRva2Vucy5zaXplID4gMCkgewogICAgYi50b2tlbnMoMCkuXzEgPT0gYmFsbG90VG9rZW5JZCAgICAgICAmJgogICAgYi5SNVtJbnRdLmdldCA9PSBTRUxGLmNyZWF0aW9uSW5mby5fMSAmJiAvLyBlbnN1cmUgdm90ZSBjb3JyZXNwb25kcyB0byB0aGlzIGJveCBieSBjaGVja2luZyBjcmVhdGlvbiBoZWlnaHQKICAgIGIuUjZbQ29sbFtCeXRlXV0uZ2V0ID09IHBvb2xPdXRIYXNoICAgICAgLy8gY2hlY2sgdmFsdWUgdm90ZWQgZm9yCiAgfSBlbHNlIGZhbHNlCiAgCiAgdmFsIGJhbGxvdEJveGVzID0gSU5QVVRTLmZpbHRlcihpc1ZhbGlkQmFsbG90KQogIAogIHZhbCB2b3Rlc0NvdW50ID0gYmFsbG90Qm94ZXMuZm9sZCgwTCwgeyhhY2N1bTogTG9uZywgYjogQm94KSA9PiBhY2N1bSArIGIudG9rZW5zKDApLl8yfSkKICAKICBzaWdtYVByb3AodmFsaWRQb29sSW4gJiYgdmFsaWRQb29sT3V0ICYmIHZhbGlkVXBkYXRlT3V0ICYmIHZvdGVzQ291bnQgPj0gbWluVm90ZXMpICAKfQ==
```

Comment:
There is an updated v2.0a version of the update box contract at https://github.com/scalahub/OraclePool/blob/58726e7751b2ff454ce4b3f45315f3cd549c2994/src/main/scala/oraclepool/v2a/Contracts.scala#L213-L270